rs/tests/testnets/perf_benchmark.rs

```diff
@@ -0,0 +1,161 @@
+//     subnet_type: application
+//   },
+//
+// To get access to P8s and Grafana look for the following lines in the ict console output:
+//
+//     prometheus: Prometheus Web UI at http://prometheus.perf_benchmarking--1692597750709.testnet.farm.dfinity.systems,
+//     grafana: Grafana at http://grafana.perf_benchmarking--1692597750709.testnet.farm.dfinity.systems,
+//     progress_clock: IC Progress Clock at http://grafana.perf_benchmarking--1692597750709.testnet.farm.dfinity.systems/d/ic-progress-clock/ic-progress-clock?refresh=10su0026from=now-5mu0026to=now,
+//
+// Happy benchmarking!
+
+use anyhow::Result;
+
+use ic_consensus_system_test_utils::rw_message::install_nns_with_customizations_and_check_progress;
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::driver::ic::{
+    AmountOfMemoryKiB, ImageSizeGiB, InternetComputer, NrOfVCPUs, Subnet, VmResources,
+};
+use ic_system_test_driver::driver::{
+    farm::HostFeature,
+    group::SystemTestGroup,
+    ic_gateway_vm::{HasIcGatewayVm, IcGatewayVm, IC_GATEWAY_VM_NAME},
+    prometheus_vm::{HasPrometheus, PrometheusVm},
+    test_env::TestEnv,
+    test_env_api::{HasTopologySnapshot, NnsCustomizations},
+};
+use std::io::{BufRead, BufReader};
+use std::process::{Command, Stdio};
+
+const PERF_HOST: &str = "dm1-dll29.dm1.dfinity.network";
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(setup)
+        .with_timeout_per_test(std::time::Duration::MAX)
+        .execute_from_args()?;
+    Ok(())
+}
+
+fn switch_to_ssd(host: &str) {
+    let script = r##"
+	#!/bin/bash
+	set -e
+        exec 2>&1
+
+        # Get the name of the currently running VM.
+	# virsh prints four lines for a single instance running, two lines of header,
+	# VM info and trailing empty line. The script should fail if the format is
+	# different or there are multiple VMs running. The trailing line is lost when
+	# assigning to a variable.
+	virsh_list=$(sudo virsh list)
+	if [ $(echo "$virsh_list" | wc -l) -ne 3 ]; then
+		echo "Unexpected virsh list output"
+		exit 2
+	fi
+	VMNAME=$(echo "$virsh_list" | awk '{ if (NR==3) print $2 }')
+
+        # Shutdown the VM
+        echo "Shutting down $VMNAME"
+	for i in {1..300}; do
+		if [ $(sudo virsh list | grep $VMNAME | wc -l) -eq 0 ]; then
+			break
+		fi
+		echo Waiting for shutdown of $VMNAME: retry $i...
+		sleep 1
+		sudo virsh shutdown $VMNAME || true
+	done
+
+        # Get the file name and dd it to disk device
+	CONFIG=$(mktemp)
+	sudo virsh dumpxml $VMNAME > $CONFIG
+	IMAGE=$(cat $CONFIG | sed -n 45p | cut -d\' -f 2)
+	echo "Moving $VMNAME to /dev/hostlvm/guest"
+	sudo dd if=$IMAGE of=/dev/hostlvm/guestos status=progress bs=512MiB oflag=direct iflag=direct
+
+        # Patch the config to point to the disk device
+	sed -i '43s/file/block/' $CONFIG
+	sed -i "44s/\/>/ discard=\'unmap\' cache=\'none\'\/>/" $CONFIG
+	sed -i "45s/file.*/dev='\/dev\/hostlvm\/guestos'\/>/" $CONFIG
```

Comment:
Here it would also be less error prone to update the XML using XPath using a tool like `xmlstarlet` (that one does need to be installed on the hosts).